internal/kaniko/execute.go

```diff
@@ -47,9 +48,94 @@ func (k *Config) Run(ctx context.Context) (err error) {
 			return err
 		}
 	}
+
+	err = k.createArtifactInfo(k.processDestinations())
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
+func (k *Config) createArtifactInfo(destinations []string) error {
+
+	for _, destination := range destinations {
+		fmt.Printf("Sending artifact info for destination: %v\n", destination)
+
+		apiUrl := os.Getenv("CLOUDBEES_API_URL")
+		if apiUrl == "" {
+			return fmt.Errorf("failed to send artifact info because of missed CLOUDBEES_API_URL variable")
+		}
+
+		apiToken := os.Getenv("CLOUDBEES_API_TOKEN")
+		if apiToken == "" {
+			return fmt.Errorf("failed to send artifact info because of missed CLOUDBEES_API_TOKEN variable")
+		}
+
+		requestURL, err := url.JoinPath(apiUrl, "/v2/workflows/runs/artifactinfos")
+		if err != nil {
+			return err
+		}
+
+		artifactInfo, err := buildCreateArtifactInfoRequest(destination)
+
+		artifactInfoBytes, err := json.Marshal(artifactInfo)
+		if err != nil {
+			return err
+		}
+
+		client := &http.Client{}
+
+		apiReq, err := http.NewRequest(
+			"POST",
+			requestURL,
+			bytes.NewReader(artifactInfoBytes),
+		)
+		if err != nil {
+			return err
+		}
+
+		apiReq.Header.Set("Authorization", fmt.Sprintf("Bearer %s", apiToken))
+		apiReq.Header.Set("Content-Type", "application/json")
+		apiReq.Header.Set("Accept", "application/json")
+
+		resp, err := client.Do(apiReq)
+		if err != nil {
+			return err
+		}
+		defer func() { _ = resp.Body.Close() }()
+
+		if resp.StatusCode != 200 {
+			return fmt.Errorf("failed to create artifact info: \nPOST %s\nHTTP/%d %s\n", requestURL, resp.StatusCode, resp.Status)
+		}
+	}
+	return nil
+}
+
+// CreateArtifactInfoObj is a map of key-value pairs that is used to store CreateArtifactInfoRequest data
+type CreateArtifactInfoObj map[string]interface{}
+
+func buildCreateArtifactInfoRequest(destination string) (CreateArtifactInfoObj, error) {
+
+	destnParts := strings.Split(destination, "/")
+	imgInfo := destnParts[len(destnParts)-1]
+	imgNameAndVersion := strings.Split(imgInfo, ":")
+	if len(imgNameAndVersion) != 2 {
+		return nil, fmt.Errorf("failed to build kaniko artifact info request: invalid destination format")
+	}
+	imgName := imgNameAndVersion[0]
+	imgVer := imgNameAndVersion[1]
+
+	artInfo := CreateArtifactInfoObj{
```

Comment:
We need `run_id` and `run_attempt` in the request payload: https://docs.google.com/document/d/1I8BUx0XpkWMiiHDz8WgDtcza8DaeEgOvqqpE94JFa9s/edit?pli=1&tab=t.0

You can get them from the context: ${{ cloudbees.run_id }}  ${{ cloudbees.run_attempt }}

Reply:
I didn't add those as we can fetch that from claims(AutomationMetadata) in JWT.

Reply:
you need to add at least the `run_id` as it is needed for the rbac check. For API calls coming from a workflow, the auth interceptor will verify that the `run_id` in the request matches the run_id in the JWT token.